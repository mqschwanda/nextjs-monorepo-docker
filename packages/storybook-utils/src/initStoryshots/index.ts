import type { StoryshotsOptions } from '@storybook/addon-storyshots/dist/api/StoryshotsOptions';
import initAddonStoryshots from '@storybook/addon-storyshots';

/**
 * Options for the `initStoryshots` function.
 */
export type InitStoryshotsOptions = StoryshotsOptions & {
  /**
   * If the snapshot tests fail when react logs a key error.
   *
   * Set this to `false` if stories avoid using the key because it would show up in the code example.
   *
   * @default true
   */
  failOnReactKeyError?: boolean
};

/**
 * A wrapper for the `@storybook/addon-storyshots` module
 *
 * See [addon storyshots docs](https://storybook.js.org/addons/@storybook/addon-storyshots) for more information.
 */
export default function initStoryshots({
  failOnReactKeyError = true,
  ...options
}: InitStoryshotsOptions) {
  if (failOnReactKeyError === false) {
    const consoleError = jest.spyOn(console, 'error');

    beforeAll(() => {
      consoleError.mockImplementation((...args) => {
        const [message] = args;
        if (message.includes('Each child in a list should have a unique "key" prop.')) {
          return;
        }

        console.error(...args); // eslint-disable-line no-console
      });
    });

    afterAll(() => {
      consoleError.mockRestore();
    });
  }

  initAddonStoryshots(options);
}
