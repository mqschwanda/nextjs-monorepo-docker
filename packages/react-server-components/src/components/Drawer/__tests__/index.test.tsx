import { render } from '@testing-library/react';
import initStoryshots from '@/__tests__/initStoryshots';
import {
  Drawer,
} from '@';

initStoryshots({
  storyKindRegex: /^@mqs\/react-server-components\/components\/Drawer$/,
});

describe('@mqs/react-server-components', () => {
  describe('components', () => {
    describe('<Drawer />', () => {
      it('renders', () => {
        const testId = Drawer.name;
        const children = 'children...';

        const { getByTestId, asFragment } = render((
          <Drawer
            className='test'
            id={testId}
            testId={testId}
          >
            { children }
          </Drawer>
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
