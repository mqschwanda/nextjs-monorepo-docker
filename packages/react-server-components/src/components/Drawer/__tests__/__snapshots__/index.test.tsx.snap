// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mqs/react-server-components components <Drawer /> renders 1`] = `
<DocumentFragment>
  <div
    class="drawer test"
    data-testid="Drawer"
    id="Drawer"
  >
    <input
      class="drawer-toggle"
      id="Drawer-toggle"
      type="checkbox"
    />
    <div
      class="drawer-content flex flex-col items-center justify-center"
    >
      <label
        class="btn btn-primary drawer-button"
        for="Drawer-toggle"
      >
        Open drawer
      </label>
    </div>
    <div
      class="drawer-side"
    >
      <label
        class="drawer-overlay"
        for="Drawer-toggle"
      />
      children...
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots @mqs/react-server-components/components/Drawer Drawer Example 1`] = `
<div
  className="drawer"
  data-testid="Drawer"
  id="DrawerExample"
>
  <input
    className="drawer-toggle"
    id="DrawerExample-toggle"
    type="checkbox"
  />
  <div
    className="drawer-content flex flex-col items-center justify-center"
  >
    <label
      className="btn btn-primary drawer-button"
      htmlFor="DrawerExample-toggle"
    >
      Open drawer
    </label>
  </div>
  <div
    className="drawer-side"
  >
    <label
      className="drawer-overlay"
      htmlFor="DrawerExample-toggle"
    />
    <ul
      className="menu menu-horizontal"
    >
      <li>
        <button
          type="button"
        >
          Navbar Item 1
        </button>
      </li>
      <li>
        <button
          type="button"
        >
          Navbar Item 2
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Drawer Drawer Responsive Example 1`] = `
<div
  className="drawer lg:drawer-open"
  data-testid="Drawer"
  id="DrawerResponsiveExample"
>
  <input
    className="drawer-toggle"
    id="DrawerResponsiveExample-toggle"
    type="checkbox"
  />
  <div
    className="drawer-content flex flex-col items-center justify-center"
  >
    <label
      className="btn btn-primary drawer-button lg:hidden"
      htmlFor="DrawerResponsiveExample-toggle"
    >
      Open drawer
    </label>
  </div>
  <div
    className="drawer-side"
  >
    <label
      className="drawer-overlay"
      htmlFor="DrawerResponsiveExample-toggle"
    />
    <ul
      className="menu menu-horizontal"
    >
      <li>
        <button
          type="button"
        >
          Navbar Item 1
        </button>
      </li>
      <li>
        <button
          type="button"
        >
          Navbar Item 2
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Storyshots @mqs/react-server-components/components/Drawer Drawer Right Side Example 1`] = `
<div
  className="drawer drawer-end"
  data-testid="Drawer"
  id="DrawerRightSideExample"
>
  <input
    className="drawer-toggle"
    id="DrawerRightSideExample-toggle"
    type="checkbox"
  />
  <div
    className="drawer-content flex flex-col items-center justify-center"
  >
    <label
      className="btn btn-primary drawer-button"
      htmlFor="DrawerRightSideExample-toggle"
    >
      Open drawer
    </label>
  </div>
  <div
    className="drawer-side"
  >
    <label
      className="drawer-overlay"
      htmlFor="DrawerRightSideExample-toggle"
    />
    <ul
      className="menu menu-horizontal"
    >
      <li>
        <button
          type="button"
        >
          Navbar Item 1
        </button>
      </li>
      <li>
        <button
          type="button"
        >
          Navbar Item 2
        </button>
      </li>
    </ul>
  </div>
</div>
`;
